recipes/chalk-to-util-styletext/src/workflow.ts

```diff
@@ -0,0 +1,161 @@
+import { getNodeRequireCalls } from "@nodejs/codemod-utils/ast-grep/require-call";
+import {
+	getNodeImportCalls,
+	getNodeImportStatements,
+} from "@nodejs/codemod-utils/ast-grep/import-statement";
+import { resolveBindingPath } from "@nodejs/codemod-utils/ast-grep/resolve-binding-path";
+import { removeLines } from "@nodejs/codemod-utils/ast-grep/remove-lines";
+import type { Edit, Range, SgNode, SgRoot } from "@codemod.com/jssg-types/main";
+import type Js from "@codemod.com/jssg-types/langs/javascript";
+
+/**
+ * Transform function that converts chalk method calls to Node.js util.styleText calls.
+ *
+ * Examples:
+ * - chalk.red("text") -> styleText("red", "text")
+ * - chalk.red.bold("text") -> styleText(["red", "bold"], "text")
+ */
+export default function transform(root: SgRoot<Js>): string | null {
+	const rootNode = root.root();
+	const edits: Edit[] = [];
+	const linesToRemove: Range[] = [];
+	const chalkBinding = "chalk";
+
+	const statements = [
+		...getNodeImportStatements(root, chalkBinding),
+		...getNodeRequireCalls(root, chalkBinding),
+		...getNodeImportCalls(root, chalkBinding),
+	];
+
+	if (!statements.length) return null;
+
+	for (const statement of statements) {
+		const binding = resolveBindingPath(statement, "$");
+
+		if (!binding) continue;
+
+		const chalkCalls = rootNode.findAll({
+			rule: {
+				any: [
+					// Pattern 1: single method calls: chalk.method(text)
+					{ pattern: `${binding}.$METHOD($TEXT)` },
+					// Pattern 2: chained method calls: chalk.method1.method2(text));
+					{
+						kind: "call_expression",
+						has: {
+							field: "function",
+							kind: "member_expression",
+						},
+					},
+				],
+			},
+		});
+
+		for (const call of chalkCalls) {
+			const methodMatch = call.getMatch("METHOD");
+			const textMatch = call.getMatch("TEXT");
+
+			if (methodMatch && textMatch) {
+				// Pattern 1: chalk.method(text) -> styleText("method", text)
+				const method = methodMatch.text();
+				const text = textMatch.text();
+				const styleMethod = COMPAT_MAP[method] || method;
```

Comment:
what if the method is valid node style thing ?

Reply:
my question is "Is there a chalk method that does not exist at all under node? " if yes what happened for the modified code ?

Reply:
I've added a test case for unsupported features: [feat(chalk-to-util-styletext): add unsupported features test](https://github.com/nodejs/userland-migrations/pull/256/commits/09e8eaf70adf1b041f781e7d652197f127c770d9) 

I'm not sure how we should warn the user though in this case. We could add a TODO comment explaining why it wasn't transformed?